sort/challenges-12.test.js

```diff
@@ -0,0 +1,372 @@
+// ------------------------------------------------------------------------------------------------
+
+const alphabetize = (strings) => {
+  return strings.sort();
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 3
+//
+// Write a function named sortByLength that takes in an array of strings and returns the same array,
+// with the strings sorted by their length, lowest to highest.
+// ------------------------------------------------------------------------------------------------
+
+const sortByLength = (strings) => {
+  return strings.sort((a, b) => {
+    if (a.length > b.length) {
+      return 1;
+    } else if (a.length < b.length) {
+      return -1;
+    } else {
+      return 0;
+    }
+  });
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 4
+//
+// Write a function named alphabetizeBetter that takes in an array of strings and returns the same
+// array, with the strings sorted alphabetically. Capitalization should not change the sort order
+// of two strings. For example, ['Alphabet', 'alphabet', 'carrot', 'Zebra'] is correctly sorted,
+// and so is ['alphabet', 'Alphabet', 'carrot', 'Zebra'].
+// ------------------------------------------------------------------------------------------------
+
+const alphabetizeBetter = (strs) => {
+  return strs.sort((a, b) => {
+    if (a.toLowerCase() > b.toLowerCase()) {
+      return 1;
+    } else if (a.toLowerCase() < b.toLowerCase()) {
+      return -1;
+    } else {
+      return 0;
+    }
+  });
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 5
+//
+// Write a function named sortByPrice that takes in an array of objects, each of which has a 'price'
+// property, and sorts those objects by price, lowest to highest, returning the same array.
+//
+// Here is an example of the input:
+// [
+//   {name: 'Sweatshirt', price: 45},
+//   {name: 'Bookmark', price: 2.50},
+//   {name: 'Tote bag', price: 15}
+// ];
+// ------------------------------------------------------------------------------------------------
+
+const sortByPrice = (objs) => {
+  return objs.sort((a, b) => {
+    if (a.price > b.price) {
+      return 1;
+    } else if (a.price < b.price) {
+      return -1;
+    } else {
+      return 0;
+    }
+  });
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 6
+//
+// Write a function named sortNumbersByLength that takes in an array of numbers and sorts those numbers
+// by their length.
+//
+// For example, [1, 14, 0.2, -281, 54782] is only correctly sorted in that order.
+// ------------------------------------------------------------------------------------------------
+
+const sortNumbersByLength = (nums) => {
+  return nums.sort((a, b) => {
+    return a.toString().length > b.toString().length;
```

Comment:
since there is only 1 line of code in your function, you could use the one line shorthand and omit the curly braces and `return`, so it would look like `return nums.sort( (a,b) => a.toString().length - b.toString().length );`